### PR TITLE
Fix testEngine causing ut to fail

### PR DESCRIPTION
**Pull request by nnsgmsone**
## What type of PR is this?

- [ ] API-change
- [X] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #8232

## What this PR does / why we need it:
* Fix testEngine causing ut to fail
* disable insert-related ut, insert's arithmetic has a memory leak that needs to be fixed specifically, #8266
* Remove unnecessary single tests and a better approach to doing this should be adopted
* Because of the special nature of logtail (two isolation levels), there is no guarantee of correctness with timestamps alone, so a physical snapshot is used to read the data

### CODE REVIEW

File: pkg/vm/engine/memoryengine/table_reader.go [hunk 1]
```diff
@@ -48,6 +48,9 @@ func (t *Table) NewReader(
 	readers []engine.Reader,
 	err error,
 ) {
+	if len(shardIDs) > 0 && len(shardIDs[0]) == 0 { // Compatible with cn's logic
```
Comment (m-schen) [inline pkg/vm/engine/memoryengine/table_reader.go:51]: add a comments that if it matches the condition, it's mem table.
or add a function like
`func IsMemTable(shardIds) {
 return
}`
to do this thing.

Comment (nnsgmsone) [reply pkg/vm/engine/memoryengine/table_reader.go]: chuli le

**Review by m-schen — APPROVED** [review]

**Review by fengttt — CHANGES_REQUESTED** [review]
You just deleted a UT, for the sake of not failing.    Once you fixed the real bug, do you plan to add the UT back?   How do we know you won't forget?

**Comment by nnsgmsone** [comment]
> You just deleted a UT, for the sake of not failing. Once you fixed the real bug, do you plan to add the UT back? How do we know you won't forget?
I added ut very early on to prevent people from writing memory leak code, but unfortunately ut never took effect because of the script, so I couldn't add back this ut until I fixed the memory leak during the execution of these sql.

**Comment by nnsgmsone** [comment]
All I can do is add some ut back in if I fix some memory leaks. That's all I can think of. Hopefully this ut will stop those memory leaks from entering the repository again @fengttt

**Review by reusee — APPROVED** [review]

**Review by fengttt — APPROVED** [review]

**Pull request merged**